packages/tailwind/src/hooks/use-tailwind.ts

```diff
@@ -0,0 +1,47 @@
+import type { Rule } from "postcss";
+import { parse } from "postcss";
+import evaluateTailwindFunctions from "tailwindcss/lib/lib/evaluateTailwindFunctions";
+import expandApplyAtRules from "tailwindcss/lib/lib/expandApplyAtRules";
+import resolveDefaultsAtRules from "tailwindcss/lib/lib/resolveDefaultsAtRules";
+import { generateRules as rawGenerateRules } from "tailwindcss/lib/lib/generateRules";
+import expandTailwindAtRules from "tailwindcss/lib/lib/expandTailwindAtRules";
+import partitionApplyAtRules from "tailwindcss/lib/lib/partitionApplyAtRules";
+import substituteScreenAtRules from "tailwindcss/lib/lib/substituteScreenAtRules";
+import { setupTailwindContext } from "../utils/tailwindcss/setup-tailwind-context";
+import type { TailwindConfig } from "../tailwind";
+import { resolveAllCSSVariables } from "../utils/css/resolve-all-css-variables";
+
+const tailwindAtRulesRoot = parse(
+  `
+  @tailwind base;
+  @tailwind components;
+`,
+).root();
+
+export function useTailwind(config: TailwindConfig) {
```

Comment:
this seems to be more of a generic function rather than a React hook, if it's not a React hook can we find a different name so the `use..` doesn't create confusion?